dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -254,6 +250,21 @@ ListenableFuture<Response> execute() {
             return result;
         }
 
+        private boolean requestCanBeRetried() {
+            return request.body()
+                    .map(body -> {
+                        if (body instanceof ConsumptionTrackingRequestBody) {
+                            return ((ConsumptionTrackingRequestBody) body).requestBodyCanBeRetried();
+                        } else {
+                            throw new SafeIllegalStateException(
+                                    "Expected a ConsumptionTrackingRequestBody but got an unexpected request body type",
+                                    SafeArg.of("bodyClassName", body.getClass()));
```

Comment:
If we implement the suggestion about only wrapping request bodies as necessary, this can become `return body.repeatable();`